src/k8s/pkg/k8sd/setup/kubelet.go

```diff
@@ -52,7 +52,7 @@ func kubelet(snap snap.Snap, hostname string, nodeIP net.IP, clusterDNS string,
 		"--fail-swap-on":                 "false",
 		"--hostname-override":            hostname,
 		"--kubeconfig":                   path.Join(snap.KubernetesConfigDir(), "kubelet.conf"),
-		"--node-labels":                  strings.Join(labels, ","),
+		"--node-labels":                  strings.Join(append(kubeletControlPlaneLabels, kubeletWorkerLabels...), ","),
 		"--read-only-port":               "0",
```

Comment:
this does not look right

Reply:
the tests for worker nodes should have caught this

Reply:
Wow good catch